### PR TITLE
Added softmax function and test to PyTorch frontend

### PR DESCRIPTION
**Pull request by Brannigan123**
Issue #2725  

Changes:
- softmax implemntation for the pytorch frontend 
- test for softmax for the pytorch frontend

### CODE REVIEW

**Comment by Oxbeefbabe** [comment]
Hi there, since PyTorch softmax doesn't take a dtype argument, the front-end function does not need to take it. It also does not need to be passed in for the test. If you remove this parameter and all the code associated with it, this should be good to go :)

**Comment by Brannigan123** [comment]
> Hi there, since PyTorch softmax doesn't take a dtype argument, the front-end function does not need to take it. It also does not need to be passed in for the test. If you remove this parameter and all the code associated with it, this should be good to go :)

Hi 👋. Thanks for the review. I had based it of the [official pytorch docs for softmax](https://pytorch.org/docs/stable/generated/torch.nn.functional.softmax.html#torch.nn.functional.softmax).

**Comment by Brannigan123** [comment]
> Hi there, since PyTorch softmax doesn't take a dtype argument, the front-end function does not need to take it. It also does not need to be passed in for the test. If you remove this parameter and all the code associated with it, this should be good to go :)

I'm making the changes requested for the implementation test.

**Comment by Oxbeefbabe** [comment]
Sorry, you are right. I seem to have found a different page on the docs. Your original implementation was correct. Apologies for the inconvenience, can you revert your code back?

**Comment by Brannigan123** [comment]
> Sorry, you are right. I seem to have found a different page on the docs. Your original implementation was correct. Apologies for the inconvenience, can you revert your code back?

Alright.Let me revert the changes.

**Comment by Oxbeefbabe** [comment]
Ok, just waiting on the Github checks to finish and if everything looks Ok this should be good to go!

**Comment by Brannigan123** [comment]
Nice. I've noticed it's taking while.

**Comment by Oxbeefbabe** [comment]
Yeah, it seems to be pretty busy today. Hopefully it wont take much longer

**Comment by Oxbeefbabe** [comment]
Ok! This looks all good to me :) Merging now

**Comment by Brannigan123** [comment]
> Ok! This looks all good to me :) Merging now

Nice. Happy to contribute.
